course5/homework6/string.cc

```diff
@@ -0,0 +1,107 @@
+#include <memory>
+#include <cstring>
+#include "string.h"
+
+namespace course{
+
+    string::string() : len(0) {}
+
+    string::string(const char* src)
+    {
+        len = std::strlen(src);
+        string_pointer = std::make_unique<char[]>(len +1);
+        std::strcpy(string_pointer.get(), src);
+    }
+
+    string::string(const string& src)
+    {
+        len = src.length();
+        string_pointer = std::make_unique<char[]>(len +1);
+        std::strcpy(string_pointer.get(), src.string_pointer.get());
+    }
+
+    string::string(string&& src)
+    {
+        len = src.length();
+        string_pointer = std::make_unique<char[]>(len +1);
```

Comment:
You are allocating a buffer here. What will happen with the buffer in the next line?